pkg/controller/testsuite/testsuite_controller.go

```diff
@@ -115,57 +128,109 @@ func (r *ReconcileTestSuite) Reconcile(request reconcile.Request) (reconcile.Res
-	if err != nil && errors.IsNotFound(err) {
-		log.Info("Creating Deployment", "namespace", deploy.Namespace, "name", deploy.Name)
-		err = r.Create(context.TODO(), deploy)
+	if r.isUninitialized(*suiteCopy) {
+		r.loggerForSuite(*suiteCopy).Info("Initialize suite")
+		testDefs, err := r.findMatchingTestDefs(*suiteCopy)
 		if err != nil {
-			return reconcile.Result{}, err
+			return reconcile.Result{}, errors.Wrapf(err, "while looking for matching test definitions for suite [%s]", suiteCopy)
 		}
-	} else if err != nil {
-		return reconcile.Result{}, err
+		currStatus, err := r.initializeTests(*suiteCopy, testDefs)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrapf(err, "while initializing tests for suite [%s]", suiteCopy)
+		}
+		suiteCopy.Status = *currStatus
+		if err := r.Client.Status().Update(ctx, suiteCopy); err != nil {
+			return reconcile.Result{}, errors.Wrapf(err, "while updating status of initialized suite [%s]", suiteCopy)
+		}
+		return reconcile.Result{Requeue: true}, nil
+	}
+	if r.isFinished(*suiteCopy) {
+		r.loggerForSuite(*suiteCopy).Info("Do nothing, suite is finished")
+		return reconcile.Result{}, nil
 	}
 
-	// TODO(user): Change this for the object type created by your controller
-	// Update the found object and write the result back if there are any changes
-	if !reflect.DeepEqual(deploy.Spec, found.Spec) {
-		found.Spec = deploy.Spec
-		log.Info("Updating Deployment", "namespace", deploy.Namespace, "name", deploy.Name)
-		err = r.Update(context.TODO(), found)
-		if err != nil {
-			return reconcile.Result{}, err
+	r.loggerForSuite(*suiteCopy).Info("Ensuring status is up-to-date")
+	updatedStatus, err := r.ensureStatusIsUpToDate(*suiteCopy)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "while ensuring status is up-to-date for suite [%s]", suiteCopy.Name)
+	}
+	suiteCopy.Status = *updatedStatus
+	pod, updatedStatus, err := r.tryScheduleTest(*suiteCopy)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "while scheduling next testing pod for suite [%s]", suiteCopy.Name)
+	}
+	if pod != nil {
+		r.loggerForSuite(*suiteCopy).Info("Testing pod created", "podName", pod.Name, "podNs", pod.Namespace)
+
+		if err := controllerutil.SetControllerReference(suiteCopy, pod, r.scheme); err != nil {
+			return reconcile.Result{}, errors.Wrapf(err, "while setting controller reference, suite [%s], pod [name %s, namespace: %s]", suiteCopy.Name, pod.Name, pod.Namespace)
 		}
+		suiteCopy.Status = *updatedStatus
 	}
-	return reconcile.Result{}, nil
+
+	if err := r.Client.Status().Update(ctx, suiteCopy); err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "while updating status of running suite", "name", suiteCopy.Name)
+	}
+
+	if pod != nil {
+		// requeue immediatelly to try schedule other tests
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	return reconcile.Result{Requeue: true, RequeueAfter: time.Second}, nil
```

Comment:
```suggestion
	return reconcile.Result{RequeueAfter: time.Second}, nil
```

source: https://github.com/kubernetes-sigs/controller-runtime/issues/303

TL;DR;

> // Implies that Requeue is true, there is no need to set Requeue to true at the same time as RequeueAfter.

Reply:
I like explicitly which is currently